packages/trpc/src/procedures/recipes/updateRecipe.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { trpcSetup, tearDown } from "../../testutils";
+import { prisma } from "@recipesage/prisma";
+import { User } from "@prisma/client";
+import type { CreateTRPCProxyClient } from "@trpc/client";
+import type { AppRouter } from "../../index";
+import { recipeFactory } from "../../factories/recipeFactory";
+
+describe("updateRecipe", () => {
+  let user: User;
+  let trpc: CreateTRPCProxyClient<AppRouter>;
+
+  beforeAll(async () => {
+    ({ user, trpc } = await trpcSetup());
+  });
+
+  afterAll(() => {
+    return tearDown(user.id);
+  });
+
+  describe("success", () => {
+    it("updates the recipe", async () => {
+      const recipe = await prisma.recipe.create({
+        data: {
+          ...recipeFactory(user.id),
+          title: "Mexican chicken",
+        },
+      });
+      const response = await trpc.recipes.updateRecipe.mutate({
+        ...recipeFactory(user.id),
+        title: "marmelad",
+        id: recipe.id,
+        labelIds: [],
+        imageIds: [],
+        folder: "main",
+      });
+      expect(response.title).toEqual("marmelad");
+    });
+  });
+  describe("error", () => {
+    it("can not update unexisting recipe", async () => {
+      return expect(async () => {
+        await trpc.recipes.updateRecipe.mutate({
+          ...recipeFactory(user.id),
+          title: "marmelad",
+          labelIds: [],
+          imageIds: [],
+          folder: "main",
+          id: "00000000-0c70-4718-aacc-05add19096b5",
+        });
+      }).rejects.toThrow("Recipe not found");
+    });
+  });
+  it("fails to update a recipe with differrent user", async () => {
+    const { user: user2 } = await trpcSetup();
+    const recipe = await prisma.recipe.create({
+      data: {
+        ...recipeFactory(user2.id),
+        title: "salad",
+      },
+    });
+
+    await expect(async () => {
+      await trpc.recipes.updateRecipe.mutate({
+        ...recipeFactory(user.id),
+        title: "salad",
+        labelIds: [],
+        imageIds: [],
+        folder: "main",
+        id: recipe.id,
+      });
+    }).rejects.toThrow("Recipe not found");
+    return tearDown(user2.id);
+  });
+  it("must throw on invalid ownership", async () => {
```

Comment:
Would suggest renaming this case to `throws an error when updating a recipe with label ids the user does not own`